src/components/ComputeModal.js

```diff
@@ -975,6 +1010,37 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
           ])
         ])
       ]),
+      !sparkMode && div({ style: { gridColumnEnd: 'span 6', marginTop: '1.5rem' } }, [
+        h(LabeledCheckbox, {
+          checked: computeConfig.autopauseEnabled,
+          disabled: !autoPauseCheckboxEnabled,
+          onChange: v => {
+            updateComputeConfig('autopauseEnabled', v)
+            // if auto pause is being turned on set threshold to default length,
+            // if auto pause is being disabled it sets threshold to 0 as this is consistent with how threshold is currently managed
+            v ? updateComputeConfig('autopauseThreshold', defaultAutoPauseLength) : updateComputeConfig('autopauseThreshold', 0)
```

Comment:
Why are we updating `autopauseThreshold` here since we don't get the user input for it until `NumberInput` below?

Reply:
On the back end, when autopause is disabled the threshold has historically been set to 0 so I thought just keeping this consistent in the front end would be a good way to go. I tried to add comments to explain this, but let me know if there is anything I can add to make this more clear or if I should not do this.

Reply:
Ah I see. The thing is, we typically want to keep React state updates to a minimum. Do you see any issues when you remove that line? I'd think it would work the same way since `autopauseThreshold` is already initialized to the default value on the top of the component and React re-renders once we `updateComputeConfig('autopauseEnabled', v)`.

Reply:
Yeah that makes sense! I'll make that change now